src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs

```diff
@@ -285,12 +285,16 @@ private void OnException(HttpContext httpContext, Exception exception)
                 var telemetry = httpContext?.Features.Get<RequestTelemetry>();
                 if (telemetry != null)
                 {
-                    telemetry.Success = false;
+                    if (int.TryParse(telemetry.ResponseCode, out int _responseCode))
+                    {
+                        telemetry.Success = !(_responseCode == 200);
```

Comment:
So if the response code is 200 then you set telemetry.Success to false but otherwise you set it to true for all other response codes?  Isn't that almost the opposite of the existing logic?